fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt

```diff
@@ -97,42 +129,56 @@ fun TabsTray(
-                                tabs = normalTabs,
-                                onTabClose = onTabClose,
-                                onTabMediaClick = onTabMediaClick,
-                                onTabClick = handleTabClick,
-                                onTabLongClick = onTabLongClick,
-                            )
+                        val showInactiveTabsAutoCloseDialog = shouldShowInactiveTabsAutoCloseDialog(inactiveTabs.size)
+                        var showAutoCloseDialog by remember { mutableStateOf(showInactiveTabsAutoCloseDialog) }
+
+                        val optionalInactiveTabsHeader: (@Composable () -> Unit)? = if (inactiveTabs.isEmpty()) {
+                            null
                         } else {
-                            TabList(
-                                tabs = normalTabs,
-                                onTabClose = onTabClose,
-                                onTabMediaClick = onTabMediaClick,
-                                onTabClick = handleTabClick,
-                                onTabLongClick = onTabLongClick,
-                            )
+                            {
+                                InactiveTabsList(
+                                    inactiveTabs = inactiveTabs,
+                                    expanded = inactiveTabsExpanded,
+                                    showAutoCloseDialog = showAutoCloseDialog,
+                                    onHeaderClick = onInactiveTabsHeaderClick,
+                                    onDeleteAllButtonClick = onDeleteAllInactiveTabsClick,
+                                    onAutoCloseDismissClick = {
+                                        onInactiveTabAutoCloseDialogCloseButtonClick()
+                                        showAutoCloseDialog = !showAutoCloseDialog
+                                    },
+                                    onEnableAutoCloseClick = {
+                                        onEnableInactiveTabAutoCloseClick()
+                                        showAutoCloseDialog = !showAutoCloseDialog
+                                    },
+                                    onTabClick = onInactiveTabClick,
+                                    onTabCloseClick = onInactiveTabClose,
+                                )
+                            }
+                        }
+
+                        if (showInactiveTabsAutoCloseDialog) {
+                            onInactiveTabsAutoCloseDialogShown()
                         }
+
+                        TabLayout(
+                            tabs = normalTabs,
+                            displayTabsInGrid = displayTabsInGrid,
+                            onTabClose = onTabClose,
+                            onTabMediaClick = onTabMediaClick,
+                            onTabClick = handleTabClick,
+                            onTabLongClick = onTabLongClick,
+                            header = optionalInactiveTabsHeader,
```

Comment:
Ah I hadn't really put it together, but this is `header` parameter is basically what I was suggesting we pass down to the `TabsTray` Composable. `optionalInactiveTabsHeader` would become a parameter instead being put together in this function. Just pointing it out though, I'm still cool with doing it as an optional follow-up later. I suppose it ultimately doesn't matter, but it does feel like that would match up with the pattern of keeping Composables dumb and passing in state/actions.

Reply:
yeah, there's definitely some Goldilocks to be found here, where Composables that are coupled from a feature perspective only get the callbacks/data flow they need and aren't necessarily all coupled to the implementation of a top-level composable.